src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -301,7 +301,11 @@ bool ciMethod::has_balanced_monitors() {
     ExceptionMark em(THREAD);
     ResourceMark rm(THREAD);
     GeneratePairingInfo gpi(method);
-    gpi.compute_map(CATCH);
+    if (!gpi.compute_map(THREAD)) {
+      // Failure is only possible for a resource-area OOM, or malformed bytecode
+      // with verification disabled.
+      assert(false, "Unexpected exception from compute_map");
```

Comment:
Since CATCH was only recently changed from fatal to assert, making these fatal() wouldn't be a change in behavior and that's what this should do.